### PR TITLE
Removed memory allocation in contact 3d + standardize documentation for state and control

### PR DESCRIPTION
**Pull request by cmastalli**

### CODE REVIEW

File: include/crocoddyl/multibody/contacts/multiple-contacts.hpp [hunk 1]
```diff
@@ -119,15 +119,15 @@ class ContactModelMultipleTpl {
    * @brief Compute the contact Jacobian and contact acceleration
    *
    * @param[in] data  Multi-contact data
-   * @param[in] x     State vector \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @@param[in] x    State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```
Comment (wxmerkt) [inline include/crocoddyl/multibody/contacts/multiple-contacts.hpp:122]:
```suggestion
   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```

File: include/crocoddyl/multibody/contacts/multiple-contacts.hpp [hunk 2]
```diff
@@ -119,15 +119,15 @@ class ContactModelMultipleTpl {
...
    */
   void calc(const boost::shared_ptr<ContactDataMultiple>& data, const Eigen::Ref<const VectorXs>& x);
 
   /**
    * @brief Compute the derivatives of the contact holonomic constraint
    *
    * @param[in] data  Multi-contact data
-   * @param[in] x     State vector \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @@param[in] x    State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```
Comment (wxmerkt) [inline include/crocoddyl/multibody/contacts/multiple-contacts.hpp:130]:
```suggestion
   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```

File: include/crocoddyl/multibody/impulses/multiple-impulses.hpp [hunk 3]
```diff
@@ -111,15 +111,15 @@ class ImpulseModelMultipleTpl {
    * @brief Compute the total contact Jacobian and contact acceleration
    *
    * @param[in] data  Multi-impulse data
-   * @param[in] x     State vector \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @@param[in] x    State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```
Comment (wxmerkt) [inline include/crocoddyl/multibody/impulses/multiple-impulses.hpp:114]:
```suggestion
   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```

File: include/crocoddyl/multibody/impulses/multiple-impulses.hpp [hunk 4]
```diff
@@ -111,15 +111,15 @@ class ImpulseModelMultipleTpl {
...
    */
   void calc(const boost::shared_ptr<ImpulseDataMultiple>& data, const Eigen::Ref<const VectorXs>& x);
 
   /**
    * @brief Compute the derivatives of the contact holonomic constraint
    *
    * @param[in] data  Multi-impulse data
-   * @param[in] x     State vector \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @@param[in] x    State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```
Comment (wxmerkt) [inline include/crocoddyl/multibody/impulses/multiple-impulses.hpp:122]:
```suggestion
   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
```

File: include/crocoddyl/multibody/contacts/multiple-contacts.hpp (hunk 1 above)
Comment (cmastalli) [reply include/crocoddyl/multibody/contacts/multiple-contacts.hpp]: done this and the others!

Thanks ;)

**Comment by cmastalli** [comment]
thanks @wxmerkt for the review!

**Pull request merged**